index.d.ts

```diff
@@ -11,28 +15,33 @@ declare class BeeQueue {
 
   on(ev: "ready",     fn: () => void): this;
   on(ev: "error",     fn: (err: Error) => void): this;
-  on(ev: "succeeded", fn: (job: BeeQueue.Job, result: any) => void): this;
-  on(ev: "retrying",  fn: (job: BeeQueue.Job, err: Error) => void): this;
-  on(ev: "failed",    fn: (job: BeeQueue.Job, err: Error) => void): this;
+  on(ev: "succeeded", fn: (job: BeeQueue.Job<T>, result: any) => void): this;
+  on(ev: "retrying",  fn: (job: BeeQueue.Job<T>, err: Error) => void): this;
+  on(ev: "failed",    fn: (job: BeeQueue.Job<T>, err: Error) => void): this;
   on(ev: "stalled",   fn: (jobId: string) => void): this;
 
   on(ev: "job succeeded", fn: (jobId: string, result: any) => void): this;
   on(ev: "job retrying",  fn: (jobId: string, err: Error) => void): this;
   on(ev: "job failed",    fn: (jobId: string, err: Error) => void): this;
   on(ev: "job progress",  fn: (jobId: string, progress: number) => void): this;
 
-  createJob<T>(data: T): BeeQueue.Job;
+  ready(): Promise<this>;
+  ready(cb?: (err: Error | null) => void): Promise<this>;
+
+  isRunning(): boolean;
 
-  getJob(jobId: string, cb: (job: BeeQueue.Job) => void): void;
-  getJob(jobId: string): Promise<BeeQueue.Job>;
+  createJob(data: T): BeeQueue.Job<T>;
```

Comment:
second change I would suggest (not very beautiful but backwards compatible):
```suggestion
  createJob<U extends T>(data: U): BeeQueue.Job<U>;
```

This will fix the following: 
```typescript
let job = this._queue.createJob<PayloadType>(payload) // Expected 0 type arguments, but got 1
```

Reply:
Would this still be a problem if T defaults to `any`?

Reply:
I don't think so. In the old version one generic may have been passed. Now no generic is allowed, so all old users how used the generic will get this tsc error:
// Expected 0 type arguments, but got 1

Reply:
Right. Thanks. Changes applied.